### PR TITLE
HW13

### PR DESCRIPTION
**Pull request by kjacks21**
My vote is to do number 6. This is pretty useful: https://www.javacodegeeks.com/2013/12/introduction-to-junit-theories.html

### CODE REVIEW

**Comment by kjacks21** [comment]
Perhaps hw5 would be a good candidate for this? https://github.com/kjacks21/swe619-fall2018/tree/master/hw5

**Comment by ngreenwald89** [comment]
6 sounds good to me.

**Comment by 1000DS000000** [comment]
Same here. and sounds good to me for using hw5.

**Comment by kjacks21** [comment]
spent some time looking at hw5 and seeing how to convert the `Theories` and I'm not sure it would be good. Thoughts? We could try putting TravisCI on this repo as part of assignment choice 5?

**Comment by 1000DS000000** [comment]
Yeah, looks like it wouldn't be. What about hw10 or would that be too similar to in class exercise?  and we could if there isn't a test we can use.

**Comment by kjacks21** [comment]
What are your thoughts on this using hw5? One thing I'm struggling with is writing a theory for throwing an exception. We could also use hw10, I have no objections to doing something similar to an in class example.

**Comment by 1000DS000000** [comment]
I'm not sure. I think its just difficult since we are using Iterators and i think Theories and Data Points are meant to be used with multiple entries like lists or arrays.

**Comment by kjacks21** [comment]
hw10 with `IntSet` it is, then?

**Comment by 1000DS000000** [comment]
yeah, maybe so. 
We could split it up like last time, each one of us can convert either the part of clone, equals or hashcode.

**Comment by kjacks21** [comment]
Messed up the message for commit f1034a2, should actually be "added hw10 files, rm hw5 files". Anyways, I made a `Theory` for `testClone`. Let me know if this is the pattern we are looking for. I figure only have one data point since we can make clones in the test methods.

**Comment by 1000DS000000** [comment]
yeah, it seems so for the datapoint, but i think to be on the safe side, we should have an array of IntSets. Also for the theories test case, i think we need to have assumptions before the rest of the code. like assuming that set is not null. 

Also, do know how we should do the story deliverable? Should we each write up about their part of the code? and maybe include a screenshot of the original test and the Theories test?
I am currently working on converting the equals method.

**Comment by kjacks21** [comment]
> yeah, it seems so for the datapoint, but i think to be on the safe side, we should have an array of IntSets.

@1000DS000000 could you make this fix for me? I'll fix the `testClone` once this fix is made.

> i think we need to have assumptions before the rest of the code. like assuming that set is not null.

Agreed. I'll make this fix.

> Also, do know how we should do the story deliverable? Should we each write up about their part of the code? and maybe include a screenshot of the original test and the Theories test?

This sounds good to me.

**Comment by kjacks21** [comment]
Updated `getClone` and the `README`. @ngreenwald89 can you do `testHashCode`?

**Comment by ngreenwald89** [comment]
@kjacks21 thanks for reminder.

File: hw13/src/IntSetTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class IntSetTest {
+
+	@Test
+	public void testOldClone() {
```
Comment (kjacks21) [inline hw13/src/IntSetTest.java:9]: Should we remove this method?

**Comment by 1000DS000000** [comment]
I'm not sure, maybe just comment it out.

**Comment by kjacks21** [comment]
We good to merge and submit? I added a portion in the README on how we used version control, which is part of part 5 of the HW.

**Comment by 1000DS000000** [comment]
i think so.

**Comment by kjacks21** [comment]
Submitted. Thanks all

**Pull request merged**